sematic/api/endpoints/runs.py

```diff
@@ -359,7 +360,12 @@ def update_run_status_endpoint(user: Optional[User]) -> flask.Response:
         run = _get_run_if_modified(run_id, future_state, jobs)
         if run is not None:
             new_future_state_value = run.future_state
-            save_run(run)
+            try:
+                save_run(run)
+            except IllegalStateTransitionError as e:
+                raise _DetectedRunRaceCondition(
```

Comment:
if you look at the code above, you will see that this results in a retry of refreshing the run state